src/App.tsx

```diff
@@ -7,33 +9,67 @@ function getRandomName(): string {
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
+type State = {
+  hasClock: boolean,
+  clockName: string,
+};
+
+export class App extends React.Component<{}, State> {
+  state: Readonly<State> = {
+    hasClock: true,
+    clockName: 'Clock-0',
+  };
+
+  clockNameId = 0;
+
+  componentDidMount() {
+    document.addEventListener('contextmenu', (e) => {
+      e.preventDefault();
+      this.addClock();
+    });
+
+    document.addEventListener('click', () => {
+      this.removeClock();
+    });
 
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    clockName = getRandomName();
-  }, 3300);
+    this.clockNameId = window.setInterval(() => {
+      this.updateClock();
+    }, 3300);
+  }
 
-  // this code stops the timer
-  window.clearInterval(timerId);
+  componentDidUpdate(_: never, prevState: State) {
+    const { hasClock, clockName } = this.state;
 
-  return (
-    <div className="App">
-      <h1>React clock</h1>
+    if (clockName !== prevState.clockName && hasClock) {
+      window.console.debug(`Renamed from ${prevState.clockName} to ${clockName}`);
+    }
+  }
 
-      <div className="Clock">
-        <strong className="Clock__name">
-          {clockName}
-        </strong>
+  componentWillUnmount() {
+    window.clearInterval(this.clockNameId);
+  }
 
-        {' time is '}
+  addClock = () => {
+    this.setState({ hasClock: false });
+  };
```

Comment:
the method `addClock` sets `hasClock` to false, but method `removeClock` sets it to true. That is counterintuitive, you should change that